### ISSUE TITLE
`Find` on a column indexed returned an unexpected warning

### ISSUE THREAD

**Issue by clun** [stargate/data-api#1573]
Given a table simple:
```console
curl --location 'http://localhost:8181//v1/default_keyspace' \
--header 'token: Cassandra:Y2Fzc2FuZHJh:Y2Fzc2FuZHJh' \
--header 'Feature-Flag-tables: true' \
--header 'Content-Type: application/json' \
--data '{
    "createTable": {
        "name": "table_simple",
        "definition": {
            "columns": {
                "email": {
                    "type": "text"
                },
                "age": {
                    "type": "int"
                },
                "name": {
                    "type": "text"
                },
                "country": {
                    "type": "text"
                },
                "human": {
                    "type": "boolean"
                }
            },
            "primaryKey": "email"
        },
         "options": {
            "ifNotExists": true
        }
    }
}'
```

- And given a simple SAI index

```console
curl --location 'http://localhost:8181//v1/default_keyspace/table_simple' \
--header 'token: Cassandra:Y2Fzc2FuZHJh:Y2Fzc2FuZHJh' \
--header 'Feature-Flag-tables: true' \
--header 'Content-Type: application/json' \
--data '{
    "createIndex": {
        "name": "idxcountry",
        "definition": {
            "column": "country",
             "options": {
                "ascii": true,
                "normalize":true,
                "caseSensitive": true
             }
        },
        "options": {
            "ifNotExists": true
        }
    }
}'
```

- Inserting a few rows

```console
curl --location 'http://localhost:8181//v1/default_keyspace/table_simple' \
--header 'token: Cassandra:Y2Fzc2FuZHJh:Y2Fzc2FuZHJh' \
--header 'Feature-Flag-tables: true' \
--header 'Content-Type: application/json' \
--data-raw '{
    "insertMany": {
        "documents": [
            {
                "email" : "cedrick@datastax.com",
                "name": "lunven",
                "age": 42, 
                "country": "France", 
                "human": true
            },
            {
                "email" : "alice@datastax.com",
                "name": "alice",
                "age": 39, 
                "country": "Australia", 
                "human": true
            },
            {
                "email" : "stefano@datastax.com",
                "name": "Lottini",
                "age": 43, 
                "country": "Italy", 
                "human": true
            },
            {
                "email" : "kavin@datastax.com",
                "name": "Gupta",
                "age": 25, 
                "country": "USA", 
                "human": true
            }
        ]
    }
}'
```

- When searching on country (indexed)
```
curl --location 'http://localhost:8181//v1/default_keyspace/table_simple' \
--header 'token: Cassandra:Y2Fzc2FuZHJh:Y2Fzc2FuZHJh' \
--header 'Feature-Flag-tables: true' \
--header 'Content-Type: application/json' \
--data '{
  "find": {
      "filter" : {"country" : "France"},
      "projection" : {}
}}
'
```

### ERROR 

- When searching on columns part of the index we got an unexpected warning

```json
{
    "data": {
        "documents": [
            {
                "country": "France",
                "name": "lunven",
                "human": true,
                "email": "cedrick@datastax.com",
                "age": 42
            }
        ],
        "nextPageState": null
    },
    "status": {
        "warnings": [
            "The filter includes columns that are not indexed. \n\nThe table default_keyspace.table_simple has the primary key: email(text).\nAnd has indexes on the columns: [None].\nThe request filtered on the un-indexed columns: country.\n\nThe query was executed without taking advantage of the primary key or indexes on the table, this can have performance implications on large tables.\n\nSee documentation at XXXX for best practices for filtering."
        ]
    }
}
```

- When searching on a field that is a BOOLEAN as a string we got an ERROR 500

```
curl --location 'http://localhost:8181//v1/default_keyspace/table_simple' \
--header 'token: Cassandra:Y2Fzc2FuZHJh:Y2Fzc2FuZHJh' \
--header 'Feature-Flag-tables: true' \
--header 'Content-Type: application/json' \
--data '{
  "find": {
      "filter" : {"human" : "true"},
      "projection" : {}
}}
'
```

- Giving the results

```json
{
    "data": {
        "documents": [],
        "nextPageState": null
    },
    "status": {
        "warnings": [
            "The filter includes columns that are not indexed. \n\nThe table default_keyspace.table_simple has the primary key: email(text).\nAnd has indexes on the columns: country.\nThe request filtered on the un-indexed columns: human.\n\nThe query was executed without taking advantage of the primary key or indexes on the table, this can have performance implications on large tables.\n\nSee documentation at XXXX for best practices for filtering."
        ]
    },
    "errors": [
        {
            "message": "Server failed: root cause: (java.lang.RuntimeException) io.stargate.sgv2.jsonapi.exception.catchable.ToCQLCodecException: Error trying to convert to targetCQLType `BOOLEAN` from value.class `java.lang.String`, value \"true\". Root cause: no codec matching value type",
            "errorCode": "SERVER_UNHANDLED_ERROR"
        }
    ]
}
```

**Comment by Yuqi-Du** [comment]
I think the first one is fixed. Need confirmation.

The second one for `Root cause: no codec matching value type` we need better error handling.

**Comment by amorton** [comment]
confirm that we are refreshing the keyspace cache in the driver before and after a DDL operation, there will still be a propagation delay between instances. This looks like the JVM did not pickup the new index, we could also look at updating the warning message to say something about schema propagation time. 

created #1744 for the codec error, solve the schema caching in this ticket

**Comment by Hazel-Datastax** [comment]
@clun Could you double-check if you could reproduce the first error now? I think the problem has been fixed by the below code in PR #1519, which was merged on the same day when you opened this issue:
```
  @Override
  public void onTableUpdated(@NonNull TableMetadata current, @NonNull TableMetadata previous) {
    // Evict from the cache because things like indexes can change for CQL Tables
    schemaCache.evictCollectionSettingCacheEntry(
        Optional.ofNullable(tenantId),
        current.getKeyspace().asInternal(),
        current.getName().asInternal());
  }
```
It maybe has a propagation delay problem, but no matter how fast I tried, I still could not reproduce the error now.
